kombu/transport/redis_cluster.py

```diff
@@ -309,41 +314,67 @@ def _brpop_start(self, timeout=1):
+            resp = self.parse_response(conn, 'BRPOP', **options)
         except self.connection_errors:
-            conn.client.close()
-            conn.client = None
             raise Empty()
-        except MovedError as e:
-            # Copied from redis-py cluster.py
-            self.client.reinitialize_counter += 1
-            if self.client._should_reinitialized():
-                self.client.nodes_manager.initialize()
-                # Reset the counter
-                self.client.reinitialize_counter = 0
-            else:
-                self.client.nodes_manager.update_moved_exception(e)
+        except MovedError:
             raise Empty()
+        conn.client.connection.send_command('BRPOP', conn.key, conn.timeout) # schedule next BRPOP
 
         if resp:
-            dest, item = resp
-            dest = bytes_to_str(dest).rsplit(self.sep, 1)[0]
-            self._queue_cycle.rotate(dest)
-            self.connection._deliver(loads(bytes_to_str(item)), dest)
+            self.deliver_response(resp)
             return True
 
     def _poll_error(self, cmd, conn, **options):
-        if cmd == 'BRPOP':
-            conn.client.parse_response(conn.client.connection, cmd, **options)
+        resp = self.parse_response(conn, 'BRPOP', **options)
+        if resp:
+            self.deliver_response(resp)
+
+        self.connection.cycle._unregister((self, self.client, conn, 'BRPOP'))
+
+    def deliver_response(self, resp):
+        dest, item = resp
+        dest = bytes_to_str(dest).rsplit(self.sep, 1)[0]
+        self._queue_cycle.rotate(dest)
+        self.connection._deliver(loads(bytes_to_str(item)), dest)
+
+    def parse_response(self, conn, cmd, **options):
+        try:
+            return conn.client.parse_response(conn.client.connection, cmd, **options)
+        except Exception as e:
+            # Copied from https://github.com/sendbird/redis-py/blob/master/redis/cluster.py#L1173
+            if e is ConnectionError or e is TimeoutError:
+                self.client.nodes_manager.startup_nodes.pop(target_node.name, None)
+                self.client.nodes_manager.initialize()
+            elif e is MovedError:
+                self.client.reinitialize_counter += 1
+                if self.client._should_reinitialized():
+                    self.client.nodes_manager.initialize()
+                    self.client.reinitialize_counter = 0
+                else:
+                    self.client.nodes_manager.update_moved_exception(e)
+            elif e is SlotNotCoveredError:
+                self.client.reinitialize_counter += 1
+                if self.client._should_reinitialized():
+                    self.client.nodes_manager.initialize()
+                    self.client.reinitialize_counter = 0
+            elif e is TryAgainError:
+                return  # try again in next BRPOP
+            elif e is AskError:
+                pass  # We should connect to other node
```

Comment:
it would be better to update node manager topology according to its response like [this code](https://github.com/sendbird/redis-py/blob/68c350520ec3ea51c6ef6ffc048b4ae2f26e5b00/redis/cluster.py#L1132C1-L1141C34) and [this](https://github.com/sendbird/redis-py/blob/68c350520ec3ea51c6ef6ffc048b4ae2f26e5b00/redis/cluster.py#L1200-L1202), since it is likely to face same issue at next trial